lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
```

Comment:
aggregates all the `:on` fields still doesn't work, I mean there are some attachments without a corresponding url in the content ( like assessment files ). 

I am thinking of doing the delta comparison, that is compare `column` and `column_was`, see what was deleted, then remove the ids deleted.